distributed/tests/test_scheduler.py

```diff
@@ -2379,34 +2383,45 @@ def test_memory():
             ]
             sleep(2)
         assert_memory(s, "managed_spilled", 1, 999)
+        # Wait for the spilling to finish. Note that this does not make the test take
+        # longer as we're waiting for recent_to_old_time anyway.
+        sleep(10)
```

Comment:
Is there something more direct we can probe here instead of sleeping for 10 seconds?

Reply:
Not really because the unmanaged memory is very volatile, so we don't know how many keys are going to be spilled out exactly. Also, as noted it doesn't slow the test down.